src/MongoDB.Driver.Core/Core/Configuration/ConnectionString.cs

```diff
@@ -747,6 +747,23 @@ private string GetHostNameForDns()
             return host;
         }
 
+#if !NETSTANDARD1_5
```

Comment:
@rstam I know you prefer `if bool` over `if !bool` : in this case though, it seems the .NET Standard 1.5 is the outlier as newer versions of .NET Standard (and surprisingly .NET Framework 4.5.2) have the needed function, so I feel like the `!NETSTANDARD1_5` is mildly more future proof. 

The alternative to `#if` would be to vendor in the `Uri.IsHexEncoding` function, which is available from the CoreFX project

Reply:
Let's not overthink it.

Once we've learned that `IsHexEncoding` is not available in .NET Standard 1.5 let's just abandon that approach and use the regular expression.